### PR TITLE
HPCC-24092 Add support for secrets to the system

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/values.yaml [hunk 1]
```diff
@@ -64,15 +64,16 @@ secrets:
...
   #Secret categories follow
-  storage: []
+  storage:
     ## Secrets that are required for accessing storage.  Currently exposed in the engines, but in the future will
     ## likely be restricted to esp (when it becomes the meta-data provider)
-    ## For example, to set the secret associated with the mystorage azure storage account use
-    ##- name: azure-mystorage
-    ##secret: storage-myazuresecret
-  ecl: []
+    ## For example, to set the secret associated with the azure stoarge account "mystorageaccount" use
```
Comment (richardkchapman) [inline helm/hpcc/values.yaml:70]: typo - stoarge

**Comment by richardkchapman** [comment]
The description in the Jira ticket does not seem to correspond to what is changed here?

**Comment by ghalliday** [comment]
That's true.  By providing secrets it allows the credentials to not be leaked, but I'll update the jira to reflect the broader fix.

**Comment by ghalliday** [comment]
@richardkchapman please review

File: ecl/eclagent/eclagent.cpp [hunk 2]
```diff
@@ -3442,6 +3442,7 @@ extern int HTHOR_API eclagent_main(int argc, const char *argv[], StringBuffer *
 #else
         setupContainerizedLogMsgHandler();
 #endif
+        setupContainerizedStorageLocations();
```
Comment (richardkchapman) [inline ecl/eclagent/eclagent.cpp:3445]: Should this be inside the if block above?

File: system/jlib/jutil.cpp [hunk 3]
```diff
@@ -3019,6 +3021,51 @@ int getEnum(const char *v, const EnumMapping *map, int defval)
...
+
+//How long can secrets be cached for?
+extern jlib_decl unsigned getSecretTimeout()
+{
+    return secretTimeoutMs;
+}
+
+extern jlib_decl void setSecretTimeout(unsigned timeoutMs)
+{
+    secretTimeoutMs= timeoutMs;
```
Comment (richardkchapman) [inline system/jlib/jutil.cpp:3038]: trivial: missing space

**Pull request merged**